### PR TITLE
Add Model Plot Predictions

### LINKED ISSUE #39: Add plotting framework for model predictions
Using `matplotlib` and/or similar libraries, create a framework for plotting to model predictions.

### LINKED ISSUE #46: Modify ini file structure
The current `.ini` file seems messy; would be better to make it more structured for parsing. Documentation is also needed.

### PR DESCRIPTION
**Pull request by KaidenFrizu**
Resolves #39 
Resolves #46 

Additions:
- Added a function to create a sequential TensorFlow model.
- Added functions in notebook for plot predictions.
- Added initializers.
- Changed regularizers.

Changes:
- Changed data structure format for `model.ini`.
- Changed subclassing layers from `tensorflow.keras.layers` to `tensorflow.keras.Model`.
- Model `.train()` method now returns loss history.

### CODE REVIEW

File: src/model.py [hunk 1]
```diff
@@ -39,31 +39,26 @@ class Encoder(tf.keras.layers.Layer):
...
+            bias_regularizer='l1',
             name='encoder_lstm',
             return_sequences=True,
         )
-        self.lstm = tf.keras.layers.Bidirectional(
-            layer=core_lstm,
-            merge_mode='sum',
+        self.bilstm = tf.keras.layers.Bidirectional(
+            layer=self.core_lstm,
+            merge_mode='concat',
```
Comment (KaidenFrizu) [inline src/model.py:60]: This might need to be added to `.ini` file arguments

Comment (KaidenFrizu) [reply src/model.py:60]: Would have to include everything else if change is needed. This is fine as is.

**Review by KaidenFrizu — COMMENTED** [review]
Need details on docstrings

File: src/plotting.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+"""
+from typing import Optional
+from typing import Union
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+class PlotPrediction:
+    """Here"""
```
Comment (KaidenFrizu) [inline src/plotting.py:24]: Add docstring

File: src/pipeline.py [hunk 3]
```diff
@@ -130,14 +132,92 @@ def _set_callbacks(self):
...
+    def plot_predict(
+        self,
+        date: str,
+        return_initial: Optional[bool] = True,
+        return_legend: Optional[bool] = True,
+        plot_title: Optional[str] = None,
+        return_ax_only: Optional[bool] = False,
+        **kwargs
+    ):
+        """Here"""
```
Comment (KaidenFrizu) [inline src/pipeline.py:177]: Add docstring details

File: src/pipeline.py [hunk 4]
```diff
@@ -130,14 +132,92 @@ def _set_callbacks(self):
...
+            return_ax_only=return_ax_only,
+            **kwargs
+        )
+
+    def transform_data(
+        self,
+        xtrain: pd.DataFrame,
+        xtest: Optional[pd.DataFrame] = None,
+    ):
+        """Here"""
```
Comment (KaidenFrizu) [inline src/pipeline.py:196]: Add docstring details

File: src/pipeline.py [hunk 5]
```diff
@@ -150,34 +230,56 @@ def model_train(
...
-            validation_data=(test_features, test_targets),
+            validation_data=(xtest, ytest),
             callbacks=self.callback_list,
             **kwargs
         )
 
+        return transformer.HistoryTransformer(hist=hist, name=self.model_name)
+
+    def predict(self, x: pd.DataFrame, **kwargs):
+        """Here"""
```
Comment (KaidenFrizu) [inline src/pipeline.py:277]: Add docstring details
